### PR TITLE
fix(react-pi): keep the effective thinking level for subscribers

### PR DESCRIPTION
**Pull request by ephraimduncan**
## Problem

In `@assistant-ui/react-pi@0.0.21`, thread subscribers can receive `xhigh` even when Pi uses `high` or `off` for the model.

The regression uses real SDK sessions with `@earendil-works/pi-coding-agent@0.84.4`. This reproduction compares the exact base source with the correction:

```sh
git clone --branch fix/pi-effective-thinking-level https://github.com/assistant-ui/assistant-ui.git
cd assistant-ui
pnpm install --frozen-lockfile
git restore --source=928c580f4132496ee6ae9dc5a64fe44ca4bfd1b7 -- packages/react-pi/src/node/ThreadSupervisor.ts
cd packages/react-pi
node node_modules/vitest/vitest.mjs run src/node/ThreadSupervisor.test.ts -t 'keeps the effective thinking level'
git restore src/node/ThreadSupervisor.ts
node node_modules/vitest/vitest.mjs run src/node/ThreadSupervisor.test.ts -t 'keeps the effective thinking level'
```

The first run fails with an extra `xhigh` event. The second run passes. No model requests occur.

## Root cause

Pi sends the effective level, but the supervisor then sends the requested level and overwrites subscriber state.

## Change

The supervisor uses the existing SDK event relay. Subscribers retain the effective level, including when another request leaves that level unchanged.

## Verification

The same two regression cases failed before the correction and passed after it. These focused checks passed:

- From `packages/react-pi`: `node node_modules/vitest/vitest.mjs run src/node/ThreadSupervisor.test.ts src/node/mapping.test.ts src/runtime/threadState.test.ts` (49 tests).
- From the repository root: `node_modules/.bin/oxlint packages/react-pi/src/node/ThreadSupervisor.ts packages/react-pi/src/node/ThreadSupervisor.test.ts`.

## Public surface

The PR includes a patch changeset for `@assistant-ui/react-pi`. Exports and signatures do not change. The existing documentation already describes the effective level.

<a href="https://cubic.dev/pr/assistant-ui/assistant-ui/pull/6894?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

<a href="https://rupic.dev/s/s2.g3_wwylpJ8n6Jw4SnI6tFuidFFosSMGcGESjAg8-LxRDUTcXUrwXw23CKMs?ref=github" target="_blank" rel="noopener noreferrer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://rupic.dev/badges/track-dark-v1.svg">Track in Rupic</picture></a>

### CODE REVIEW

File: packages/react-pi/src/node/ThreadSupervisor.test.ts [hunk 1]
```diff
@@ -179,6 +184,67 @@ describe("PiThreadSupervisor", () => {
     expect(session.setThinkingLevel).toHaveBeenCalledTimes(2);
   });
 
+  it.each([
+    { provider: "anthropic", modelId: "claude-opus-4-5", levels: ["high"] },
+    { provider: "openai", modelId: "gpt-4o", levels: [] },
```
[the remark this answers is not in the document]
Comment (ephraimduncan) [reply packages/react-pi/src/node/ThreadSupervisor.test.ts]: The test selects models by thinking capabilities instead of catalog names. It keeps the literal high/off expectations and the repeated-request cases. With Pi SDK 0.84.4, all 85 focused tests passed. The same regression cases failed against the base source with extra xhigh events.

**Review by okisdev — APPROVED** [review]
approved. the double emit was the whole of it: `setThinkingLevel` on the pi side clamps first and emits `thinking_level_changed` with the effective level itself, and only when the level actually moves, so the supervisor's own emit landed after the relay carrying the requested value and left subscribers disagreeing with `snapshotOf`, which has always read `session.thinkingLevel`. deleting it fixes the seam instead of reconciling two writers, and the README already promised the effective level, so the code was the thing that was wrong.

i ran both sides in a worktree against the pinned SDK rather than reading the patch: with `ThreadSupervisor.ts` restored to 928c580f4 both regression cases fail (`expected [ 'xhigh' ] to deeply equal []`), and on the head the three focused files pass 49 tests. i also checked the new test cannot reach the network or the user's pi config (`modelsPath: null` plus `refreshOnCreate: false`, and a refresh would additionally need `allowModelNetwork: true`; authPath, agentDir and the settings manager are all temp or in memory), and that the capability predicate now matches 39 models out of the 1290 in the builtin catalog, which is what closes the catalog coupling from the first round. worth knowing for later: `xhigh === null` in that predicate doubles as the `thinkingLevelMap` existence guard, so it has to stay strict even though `max` beside it is `== null`.

the one behavior change that stands is that a request which leaves the effective level unchanged now emits nothing at all. it does not block: `ThreadController.setThinkingLevel` refreshes after the call and the example picker is driven entirely by `metadata.config.thinkingLevel`, so client state still converges. the deleted comment is the only loose end, since `clearQueue` two methods up documents exactly this "the session emits its own event, the relay forwards it" arrangement and `setThinkingLevel` now relies on it silently.
